backend/wallet/services/transaction.py

```diff
@@ -124,16 +136,58 @@ def process_incoming_transaction(
             return
 
         raise InvalidTravelRuleMetadata(
-            f"Travel Rule metadata decode failed: Transaction {transaction_id} with reference ID {reference_id} "
+            f"Travel Rule metadata decode failed: Transaction {transaction.id} with reference ID {reference_id} "
             f"should have amount: {transaction.amount}, status: {TransactionStatus.OFF_CHAIN_READY}, "
             f"source address: {transaction.source_address}, destination address: {transaction.destination_address},"
             f" but received transaction has amount: {amount}, status: {transaction.status}, "
             f"source address: {sender_address}, destination address: {receiver_address}"
         )
 
-    if not receiver_id:
-        log_execution("Incoming transaction had no metadata. crediting inventory")
-        receiver_id = get_account(account_name=INVENTORY_ACCOUNT_NAME).id
+    # Could not find receiver for the transaction, so send back a refund from inventory account
+    if receiver_id is None:
+        sender_id = get_account(account_name=INVENTORY_ACCOUNT_NAME).id
+        send_transaction(
+            sender_id=sender_id,
+            amount=amount,
+            currency=currency,
+            destination_address=sender_address,
+            destination_subaddress=sender_subaddress,
+            payment_type=TransactionType.REFUND,
+            referenced_event=referenced_event,
+        )
+
+    if referenced_event:
+        original_txn = get_transaction_by_referenced_event(referenced_event)
+        if original_txn is None:
+            raise InvalidRefund(
```

Comment:
probably should not raise error here, mark the txn as invalid refund instead, because the txn is already onchain, raise error here won't change the fact you received the refund.